cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -298,6 +302,21 @@ protected void getLiveInfo(HttpRequest request, HttpResponder responder, String
     }
   }
 
+  // Adds a schedule to the scheduler as well as to the appspec
+  protected void addSchedule(Scheduler scheduler, Store store, Id.Program programId,
+                             ScheduleSpecification scheduleSpecification) throws OperationException {
+    scheduler.schedule(programId, scheduleSpecification.getProgram().getProgramType(),
```

Comment:
Added unit test case. Addition of the new schedule with the same name should throw and exception.